### PR TITLE
Adds category folders

### PR DESCRIPTION
**Pull request by schalkventer**
image

image

### CODE REVIEW

**Comment by adieyal** [comment]
Very nice

File: js/views/add_story.js [hunk 1]
```diff
@@ -13,37 +13,68 @@ var AddStoryView = Backbone.View.extend({
...
+      return activeTopics;
     }
 
-    this.stickit();
+    if (this.model instanceof CategoriesList) {
+      props.categoriesList = this.model.toJSON();
+      
+    } else if (this.model instanceof Category) {
+      props.category = getActiveTopics(this.model);
+      props.title = this.model.toJSON().name;
```
Comment (longhotsummer) [inline js/views/add_story.js]: Does this need to be localised?

File: js/views/add_story.js [hunk 2]
```diff
@@ -13,37 +13,68 @@ var AddStoryView = Backbone.View.extend({
...
 
   initialize: function(options) {
-    this.model = new Story({}, {parse: true});
-    if (options.topic) this.model.set('topic', options.topic);
+    if (options.category && options.topic) {
+      this.model = new Story({}, {parse: true});
+      this.model.set('topic', options.topic);
 
+    } else if (options.category) {
+      this.model = _.find(PocketReporter.categoriesList.models, function(item) {
```
Comment (longhotsummer) [inline js/views/add_story.js]: I think you can use `PocketReporter.categoriesList.get(options.category)`.

File: js/views/add_story.js [hunk 3]
```diff
@@ -13,37 +13,68 @@ var AddStoryView = Backbone.View.extend({
...
     }
 
-    this.stickit();
+    if (this.model instanceof CategoriesList) {
+      props.categoriesList = this.model.toJSON();
+      
+    } else if (this.model instanceof Category) {
+      props.category = getActiveTopics(this.model);
+      props.title = this.model.toJSON().name;
+      props.baseUrl = '/#add/' + this.model.toJSON().id + '/';
```
Comment (longhotsummer) [inline js/views/add_story.js]: `this.model.get('id')`

File: js/views/add_story.js [hunk 4]
```diff
@@ -13,37 +13,68 @@ var AddStoryView = Backbone.View.extend({
...
+      story: null,
+      baseUrl: '/#add/',
+    }
 
-    // translate
-    _.each(topics, function(t) {
-      t.name = PocketReporter.polyglot.t('topics.' + t.id + '.name');
-    });
+    function getActiveTopics(model) {
+      var activeTopicsKeys = model.toJSON().topics;
```
Comment (longhotsummer) [inline js/views/add_story.js]:
```js
var activeTopics = _.map(category.get('topics'), function(id) { return PocketReporter.topics.get(id); });
```

**Pull request merged**